src/Money.php

```diff
@@ -56,8 +56,13 @@
      */
     public function __construct($amount, Currency $currency)
     {
-        if (filter_var($amount, FILTER_VALIDATE_INT) === false && !Number::fromString($amount)->isInteger()) {
-            throw new \InvalidArgumentException('Amount must be an integer(ish) value');
+        if (filter_var($amount, FILTER_VALIDATE_INT) === false) {
+            $numberFromString = Number::fromString($amount);
+            if (!$numberFromString->isInteger()) {
+                throw new \InvalidArgumentException('Amount must be an integer(ish) value');
+            }
+
+            $amount = $numberFromString->getIntegerPart();
```

Comment:
I am still failing to see how this is a solution. Can you please enlighten me, because I am missing something. isInteger checks if the fractional part is empty, and getIntegerPart returns the integer part. But if the fractional part is empty, $amount itself should be valid as well, shouldn't it?